atd-etl/app/process/helpers_import.py

```diff
@@ -13,35 +13,58 @@
 import json
 import re
 import datetime
-import web_pdb
 
 # Dependencies
 from .queries import search_crash_query, search_crash_query_full
 from .request import run_query
 from .helpers_import_fields import CRIS_TXDOT_FIELDS, CRIS_TXDOT_COMPARE_FIELDS_LIST
 
 
-def generate_template(name, function, fields):
+def generate_template(name, function, fields, fieldnames=[], upsert=False, constraint=""):
     """
     Returns a string with a graphql template
-    :param name:
-    :param function:
-    :param fields:
-    :return:
-    """
+    :param str name: The name of the graphql mutation
+    :param str function: The name of the graphql function
+    :param str fields: The value of the fields in graphql expression
+    :param str[] fieldnames: An array of strings containing the names of the columns
+    :param bool upsert: If true, adds upsert logic; false otherwise.
+    :param str constraint: The name of the constraint on_conflict
+    :return str:
+    """
+    if upsert:
```

Comment:
Thanks for opening my eyes to upserts in Hasura. I read the docs a little bit just now, and this is making me understand how you are controlling which fields will update even better. 👍